src/apint/utils.rs

```diff
@@ -392,10 +392,10 @@ impl ApInt {
     #[inline]
     pub fn is_one(&self) -> bool {
         match self.access_data() {
-            DataAccess::Inl(digit) => digit == Digit::one(),
+            DataAccess::Inl(digit) => digit == Digit::ONE,
             DataAccess::Ext(digits) => {
                 let (last, rest) = digits.split_last().unwrap_or_else(|| unreachable!());
-                last.is_one() && rest.iter().all(|digit| digit.is_zero())
```

Comment:
tbh I prefer the `is_one` version more because it is more readable imo.

Reply:
I guess I can keep `is_one` as a method for internal `Digit` use. `Digit::ONE` is used often enough that it could stay.

I am thinking about removing all `is_one` and creation of value 1 `ApInt`s. The reason for this is the corner case where single bit signed interpretation `ApInt`s can only represent negative one and zero. I thought about using this documentation:
```
    /// Returns `true` if the **unsigned** interpretation `ApInt` represents the value one (`1`).
    ///
    /// # Corner Case
    ///
    /// Normally, both signed and unsigned `ApInt`s have the same representation for the value one.
    /// However, the most significant and least significant bits are the same bit for
    /// `ApInt`s with a bitwidth of 1. This means that signed `ApInt`s with bitwidth 1
    /// can only represent negative one and zero. This function treats the
    /// `ApInt` as unsigned in this corner case.
    /// 
    /// # Note
    ///
    /// - One (`1`) is also called the multiplicative neutral element.
    /// - This operation is more efficient than comparing two instances of `ApInt`
```
The corner case problems stack up when considering what `Int` will do. Single bit `ApInt`s with a set bit also fall under the more general case of `ApInt::signed_min_value()`. I think I should add some documentation on the `ApInt` struct or very high up about being careful with `ApInt::signed_min_value()` in algorithms and mentioning single bit `ApInt`s.

I think that zero, unsigned_min_value, unsigned_max_value, signed_min_value, and signed_max_value are the only special numbers that get their own functions.

Should I remove all public facing special `is_one` now?

Reply:
I think in the documentation I could give an example of the proper way to handle magic numbers.
For example, handling creation of 1 can be handled `ApInt::unsigned_max_value(bitwidth 1).unsigned_extension_fn...` or `ApInt::from(123u8).extend...` which forces the user to consider what happens with small bitwidth values